store-and-retrieve-data/index.md

```diff
@@ -5,55 +5,57 @@ description: Learn about the complete data storage and retrieval process in Data
 
 # Store and Retrieve Data Overview
 
-DataHaven is a verifiable storage network that separates storage from verification. Storage providers hold your file bytes off-chain, while the chain records compact on-chain commitments (“receipts”) so integrity can be checked at read time.
+DataHaven is a verifiable storage network that separates storage from verification. Storage providers hold your file bytes off-chain, while the chain records compact on-chain commitments. These commitments serve as "receipts" so data can be checked at read time.
 
-This section provides a high-level overview of how to create a bucket, upload, and retrieve files.
+## Understand Core Workflow
 
-## How File Storage Works at a Glance
+The steps of the core workflow are as follows:
 
-1. **Pick an MSP and bucket**: Connect your wallet or app identity. Select a Main Storage Provider (MSP) and create or reuse a bucket. Today, access is owner-only; sharing/public access is planned. Set your desired replication factor (BSP replicas).
+```mermaid
+graph LR
+    A[Connect to MSP,<br>Create bucket,<br>Set BSP replication factor] --> B[Issue<br>storage<br>request]
+    B --> C[Verify<br>registration<br>on-chain]
+    C --> D[Upload file<br>to MSP]
+    D --> E[Retrieve<br>your<br>data]
+```
 
-2. **Upload data**: Your app opens a storage request to the MSP. The MSP accepts it, stores the bytes off-chain, and coordinates replication to protocol-assigned Backup Storage Providers (BSPs). If enough BSPs acknowledge within the window, the MSP anchors a per-bucket update on-chain (a compact commitment that indirectly commits to the file). If not, the request expires: the MSP may still have the bytes, but the file is not yet fully secured by the network.
-
-3. **Retrieve data**: If the storage request is fulfilled (replication reached the target and the bucket’s root was anchored), the MSP returns the bytes plus a Merkle proof and your app verifies it against the bucket’s on-chain commitment. If the request is pending or expired, the MSP may still serve the bytes, but there’s no on-chain anchor to verify against. Treat it as not yet secured until replication completes.
-
-Ready to try it in the dapp? It’s a straightforward create, upload, and retrieve process.
+## Key Considerations
 
-## Start Here: Core Workflows
+As you plan your development, consider the following:
 
-- Create a bucket and choose an MSP. Access is currently owner-only; sharing/public access is on the roadmap. For confidentiality, you can encrypt files on the client side before uploading.
-- Upload files: the app creates a storage request; once required BSP replication acks arrive, the MSP anchors a per-bucket update on-chain that commits to your file(s).
-- Retrieve files later with a lightweight integrity check against the bucket’s on-chain commitment (only when fulfilled).
+- **Bucket access**: Today, access is owner-only. Sharing and public access are planned.
```

Comment:
I don't why but "Today" is triggering to me haha, it makes this sentence jump out more as not future-proof. Maybe "At time of writing" or something?